bundles/at.medevit.elexis.agenda.ui/src/at/medevit/elexis/agenda/ui/composite/AppointmentDetailComposite.java

```diff
@@ -199,16 +222,34 @@ public void proposalAccepted(IContentProposal proposal) {
 				txtPatSearch.setData(prop.getIdentifiable());
 				appointment.setSubjectOrPatient(prop.getIdentifiable().getId());
 				refreshPatientModel();
+				updateEmailControlsStatus();
 			}
 		});
-		txtPatSearch.addModifyListener(new ModifyListener() {
-			@Override
-			public void modifyText(ModifyEvent e) {
-				reloadContactLabel();
-				if (!txtDataIsMatchingContact()) {
-					appointmentsViewer.setInput(Collections.emptyList());
-				}
+		txtPatSearch.addModifyListener(e -> {
+		    reloadContactLabel();
+		    if (!txtDataIsMatchingContact() || StringUtils.isBlank(txtPatSearch.getText())) {
+		        txtPatSearch.setData(null);
+				ContextServiceHolder.get().setActivePatient(null);
+		    }
+			if (!txtDataIsMatchingContact()) {
+				appointmentsViewer.setInput(Collections.emptyList());
 			}
+		    updateEmailControlsStatus();
+		    Object data = txtPatSearch.getData();
+		    if (data instanceof IPatient) {
+		        IPatient selectedPatient = (IPatient) data;
+				if (selectedPatient != null) {
+					ContextServiceHolder.get().setActivePatient(selectedPatient);
+				}
+		    }
```

Comment:
Weshalb wird hier der selektierte Patient in Elexis gesetzt?

Das Verändert den workflow in Elexis. Ist das Absicht?